src/Stratis.Bitcoin.IntegrationTests.Common/EnvironmentMockUpHelpers/CoreNode.cs

```diff
@@ -120,7 +118,11 @@ public CoreNode NoValidation()
         /// <returns>This node.</returns>
         public CoreNode SetConnectInterceptor(Action<ChainedHeaderBlock> interceptor)
         {
-            this.builderConnectInterceptor = interceptor;
+            this.FullNode.NodeService<ISignals>().OnBlockConnected += delegate(ChainedHeaderBlock block)
```

Comment:
@noescape00 My concern here is that everytime we start a node we will be adding to the `OnBlockConnected` event handler, whereas previously it was just a singleton? What are your thoughts on a possible memory leak with the tests. Or doesn't it matte, as it gets disposed with the node's disposal.

Reply:
It doesn't matter. If it would we would dispose things in the tests. 
All of it will just be consumed by the garbage collector.

Reply:
👍